source/components/composing-components.md

```diff
@@ -0,0 +1,108 @@
+Components really shine when you use them to their full potential, which is when you compose them. Take for example the `<ul>` element, and the fact that only `<li>` elements are appropriate as children. If we want the same type of behavior, then we have to compose our components.
+
+Just like we compose regular HTML elements, we can do the same with components.
+
+```hbs
+{{#user-list sort-by='name' as |user|}}
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+### Component Blocks
+
+Components can be used in two forms, just like regular HTML elements.
+
+Inline Form
+```hbs
+{{user-list}}
+```
+
+Block Form
+```hbs
+{{#user-list}}
+  {{!-- custom template here --}}
+{{/user-list}}
+```
+
+To compose components, we must use the block form, but we must also
+be able to distinguish from within our component which form the user
+is implementing. This can be done with the `template` attribute.
+
+```hbs
+{{#if template}}
+  {{yield}}
+{{else}}
+  <p>No Template Specified</p>
+{{/if}}
+```
+
+We can check if `template` is truthy, and if it is that means that the user specified a custom template. Well, once we have a template, we probably want to use that in our component, and that's exactly what `{{yield}}` does.
+
+This helper can be used once, or many times. You can make your component into a type of `<ul>` element, that is a list that will repeat n times. Like the following example, where we can output
+a custom summary.
+
+```hbs
+{{#each item in data}}
+  <h3>{{item.title}}</h3>
+  <p>{{yield}}</p>
+{{/each}}
+```
+
+Which can be used like so:
+
+```hbs
+{{#post-list}}
```

Comment:
Need to bind in some data otherwise it's unclear where those titles are coming from below...